### PR TITLE
Fixed #24089 -- Added check for when ModelAdmin.fieldsets[1]['fields'] isn't a tuple.

### PR DESCRIPTION
**Pull request by arcturusannamalai**
Change to check for tuple/list on the label 'fields' of fieldsets field of ModelAdmin.
Ticket 24089

### CODE REVIEW

File: django/contrib/admin/checks.py [hunk 1]
```diff
@@ -130,7 +130,10 @@ def _check_fieldsets_item(self, cls, model, fieldset, label):
                     id='admin.E011',
                 )
             ]
-
+        elif not isinstance(fieldset[1]['fields'],(list,tuple)):
+            # user missed the ',' following field name tuple probably
+            return must_be('a list or tuple',option=label, obj=cls,id='admin.E008')
```
Comment (collinanderson) [inline django/contrib/admin/checks.py]: Hi, thanks for the patch. Please use proper pep8 spacing:

`elif not isinstance(fieldset[1]['fields'], (list, tuple)):`
`return must_be('a list or tuple', option=label, obj=cls, id='admin.E008')`

File: tests/admin_checks/tests.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def check(cls, model, **kwargs):
         return ['error!']
 
 
+
```
Comment (charettes) [inline tests/admin_checks/tests.py]: Only two empty lines required. You can run `flake8` on source to spot these kind of warnings.

**Comment by arcturusannamalai** [comment]
Thanks for reviews @charettes, @timgraham  and @collinanderson. Here is my second cleanup, tests, attempt.

**Comment by charettes** [comment]
Buildbot, test this please.

**Comment by arcturusannamalai** [comment]
Verified test locally.

**Comment by charettes** [comment]
Buildbot, test this please.

**Comment by arcturusannamalai** [comment]
Buildbot, test this please.

File: .gitignore [hunk 3]
```diff
@@ -1,11 +1 @@
...
-*.py[co]
-__pycache__
-MANIFEST
-dist/
-docs/_build/
-docs/locale/
-tests/coverage_html/
-tests/.coverage
-build/
+tests/test.sh
```
Comment (berkerpeksag) [inline .gitignore]: Looks like you accidentally committed this.

File: tests/admin_checks/tests.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ class ExtraFieldForm(SongForm):
         return ExtraFieldForm
 
     fieldsets = (
-        (None, {
+        ('administration', {
```
Comment (berkerpeksag) [inline tests/admin_checks/tests.py]: Why did you change this line? It looks unnecessary to me.

File: tests/admin_checks/tests.py [hunk 5]
```diff
@@ -124,6 +124,29 @@ def test_custom_get_form_with_fieldsets(self):
...
+            list_editable = ["title"]
+            fieldsets = [
+                (None, {
+                    "fields": ("title")  # expect a comma
+                }),
+            ]
+        errors = NotATupleAdmin.check(model=Song)
+        expected = [
+            checks.Error(
+                "The value of 'fieldsets[0]' must be a list or tuple.",
```
Comment (berkerpeksag) [inline tests/admin_checks/tests.py]: If I'm reading the code correctly, `fieldsets[0]` is a tuple. Perhaps the error message should be `"The value of 'fieldsets[1]['fields']' must be a list or tuple."`

**Comment by berkerpeksag** [comment]
Please squash the commits when you're done. See also https://docs.djangoproject.com/en/dev/internals/contributing/committing-code/#committing-guidelines for the correct commit message format.

File: tests/admin_checks/tests.py (hunk 5 above)
Comment (arcturusannamalai) [inline tests/admin_checks/tests.py]: OK. Let me rework it. Thanks for pointers.

**Comment by timgraham** [comment]
Could you also add a test for this problelm in a non-first fieldset? That seems to be fixed already, but I don't see an existing test for it and wasn't sure if it worked.
```diff
diff --git a/tests/admin_checks/tests.py b/tests/admin_checks/tests.py
index 93b5018..8a51a56 100644
--- a/tests/admin_checks/tests.py
+++ b/tests/admin_checks/tests.py
@@ -126,11 +126,9 @@ class SystemChecksTestCase(TestCase):
 
     def test_fieldsets_tuple(self):
         """
-        Tests for basic tuple/lists within field sets (#24089)
+        Tests for tuple/lists within the first fieldset (#24089)
         """
         class NotATupleAdmin(admin.ModelAdmin):
-            list_display = ["pk", "title"]
-            list_editable = ["title"]
             fieldsets = [
                 (None, {
                     "fields": ("title")  # expect a comma
@@ -147,6 +145,30 @@ class SystemChecksTestCase(TestCase):
         ]
         self.assertEqual(errors, expected)
 
+    def test_fieldsets_tuple(self):
+        """
+        Tests for tuple/lists within the second fieldset
+        """
+        class NotATupleAdmin(admin.ModelAdmin):
+            fieldsets = [
+                (None, {
+                    "fields": ("title",)
+                }),
+                ('foo', {
+                    "fields": "author"  # not a tuple
+                }),
+            ]
+        errors = NotATupleAdmin.check(model=Song)
+        expected = [
+            checks.Error(
+                "The value of 'fieldsets[1]' must be a list or tuple.",
+                hint=None,
+                obj=NotATupleAdmin,
+                id='admin.E008',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
     def test_exclude_values(self):
         """
         Tests for basic system checks of 'exclude' option values (#12689)
```

**Comment by arcturusannamalai** [comment]
I updated all changes requested by @timgraham  and @berkerpeksag . Hopefully this is sufficient.
https://github.com/arcturusannamalai/django/commit/93c40f2d57cbb5f7c655682db369c772ed48afce

**Comment by collinanderson** [comment]
buildbot, test this please

**Comment by timgraham** [comment]
merged in b75c707943e159b80c179c538721406bbfb8b120, thanks.
